crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -133,134 +133,129 @@ async fn app_validation_workflow_inner(
     );
     let sleuth_id = conductor.config.sleuth_id();
 
-    // Build an iterator of all op validations
-    let iter = sorted_ops.into_iter().map({
-        let network = network.clone();
-        let workspace = workspace.clone();
-        move |so| {
-            let network = network.clone();
-            let conductor = conductor.clone();
-            let workspace = workspace.clone();
-            let validation_dependencies = validation_dependencies.clone();
-            let dna_hash = dna_hash.clone();
-            async move {
-                let (op, op_hash) = so.into_inner();
-                let op_type = op.get_type();
-                let action = op.action();
-                let dependency = op.sys_validation_dependency();
-                let op_lite = op.to_lite();
-
-                // If this is agent activity, track it for the cache.
-                let activity = matches!(op_type, DhtOpType::RegisterAgentActivity).then(|| {
-                    (
-                        action.author().clone(),
-                        action.action_seq(),
-                        dependency.is_none(),
-                    )
-                });
-
-                // Validate this op
-                let dht_op_hash = op.to_hash();
-                let cascade = Arc::new(workspace.full_cascade(network.clone()));
-                let validation_outcome = match dhtop_to_op(op, cascade).await {
-                    Ok(op) => {
-                        let validation_dependencies = validation_dependencies.clone();
-                        validate_op_outer(
-                            dna_hash,
-                            &op,
-                            &dht_op_hash,
-                            &conductor,
-                            &workspace,
-                            &network,
-                            validation_dependencies,
-                        )
-                        .await
-                    }
-                    Err(e) => Err(e),
-                };
-
-                (op_hash, dependency, op_lite, validation_outcome, activity)
+    let cascade = Arc::new(workspace.full_cascade(network.clone()));
+    let validation_dependencies = validation_dependencies.clone();
+    let accepted_ops = Arc::new(Mutex::new(0));
+    let awaiting_ops = Arc::new(Mutex::new(0));
+    let rejected_ops = Arc::new(Mutex::new(0));
```

Comment:
Could have been `AtomicU32`?

Reply:
:+1:

Reply:
I tried that first but couldn't clone it when moved into the database write closure. Now as a combo with arc it's fine.

Reply:
And don't know what difference it makes =D